test/e2e/workload_test.go

```diff
@@ -0,0 +1,301 @@
+package e2e_test
+
+import (
+	"github.com/appscode/go/types"
+	"github.com/appscode/stash/apis"
+	api "github.com/appscode/stash/apis/stash/v1alpha1"
+	"github.com/appscode/stash/apis/stash/v1beta1"
+	"github.com/appscode/stash/pkg/util"
+	"github.com/appscode/stash/test/e2e/framework"
+	"github.com/appscode/stash/test/e2e/matcher"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+)
+
+var _ = Describe("Deployment", func() {
+	var (
+		err              error
+		f                *framework.Invocation
+		cred             core.Secret
+		deployment       apps.Deployment
+		recoveredDeployment apps.Deployment
+		repo             *api.Repository
+		backupCfg        v1beta1.BackupConfiguration
+		restoreSession   v1beta1.RestoreSession
+		pvc              *core.PersistentVolumeClaim
+		targetref		 v1beta1.TargetRef
+	)
+	var(
+		sampleData []string
+		restoredData []string
+	)
+	BeforeEach(func() {
+		f = root.Invoke()
+	})
+	JustBeforeEach(func() {
+		cred = f.SecretForLocalBackend()
+		if missing, _ := BeZero().Match(cred); missing {
+			Skip("Missing repository credential")
+		}
+		pvc = f.GetPersistentVolumeClaim()
+		err = f.CreatePersistentVolumeClaim(pvc)
+		Expect(err).NotTo(HaveOccurred())
+		repo = f.Repository(cred.Name, pvc.Name)
+
+		backupCfg = f.BackupConfiguration(repo.Name, targetref)
+		restoreSession = f.RestoreSession(repo.Name, targetref)
+	})
+	AfterEach(func() {
+		err = f.DeleteSecret(cred.ObjectMeta)
+		Expect(err).NotTo(HaveOccurred())
+		err = framework.WaitUntilSecretDeleted(f.KubeClient, cred.ObjectMeta)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	var (
+		shouldBackupNewDeployment = func() {
```

Comment:
Rename `shouldBackupNewDeployment` to `runDeploymentBackup` or `testDeploymentBackup`